File: solutions/fsharp/cars-assemble/1/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int) : float =
    match speed with
    | 0 -> 0.0
    | 1
    | 2
    | 3
    | 4 -> 1.0
    | 5
    | 6
    | 7
    | 8 -> 0.9
    | 9 -> 0.8
    | 10 -> 0.77
    | _ -> 0

let productionRatePerHour (speed: int) : float =
    float (221 * speed) * successRate (speed)

let workingItemsPerMinute (speed: int) : int =
    int (productionRatePerHour (speed) / float 60)

File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        var succ = 1.0;
        switch (speed) {
            case 5:
            case 6:
            case 7:
            case 8:
                succ = 0.9;
                break;
            case 9:
                succ = 0.8;
                break;
            case 10:
                succ = 0.77;
                break;
        }
        return 221.0 * ((double) speed) * succ;
    }

    public int workingItemsPerMinute(int speed) {
        return (int)( productionRatePerHour(speed) / 60);
    }
}
